go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -149,6 +151,16 @@ func Serve(
 	if err != nil {
 		return err, nil
 	}
+
+	// check if user exists, else add them as superuser; might need to do something else for sql-server
+	if user := sqlEngine.GetUnderlyingEngine().Analyzer.Catalog.MySQLDb.GetUser(config.ServerUser, config.ServerHost, false); user == nil {
+		sqlEngine.GetUnderlyingEngine().Analyzer.Catalog.MySQLDb.AddSuperUser(config.ServerUser, config.ServerPass)
+	}
```

Comment:
This should be a bit different. Refer to the bullet points on my comment:
* For both `dolt sql` and `dolt sql-server`, if the user given via `--user` exists then we assume that user. What this means for `dolt sql-server` is that, since users are attached to connecting clients, the `--user` directive is just ignored if the user exists. This would also mean that `--password` is ignored for existing users.
* If no user is specified for `dolt sql-server` and the privilege file does not exist then we create the default `root` super user. If a privilege file exists then we only create a super user if one was specified through `--user`. This prevents the situation where an admin is unaware that a root super user exists that others can log into (assuming they can connect as `localhost`).

And just to clarify it a bit: the `--user` argument should only be creating super users. If the user already exists, then the argument is ignored. The default value of `root` is **only** used when no privilege file exists, otherwise we require an explicit `--user` argument to construct a super user.